lib/graphite_async.ml

```diff
@@ -219,9 +219,14 @@ let add_percentile_observation_opt t ~key value =
   | Some t -> add_percentile_observation t ~key value
 
 module Result = struct
-  let time ?graphite ~key ~f v =
+  let time ?graphite ?callback ~key ~f v =
     let timer = timer_ms () in
     let%bind r = f v in
+    let time_span = timer () in
+    let () = match callback with
+      | Some callback -> callback time_span v r
+      | None -> ()
+    in
```

Comment:
Consider extracting the snippet above into a helper function.

Reply:
I made it into a one-liner. Not sure if that serves the purpose.